### ISSUE TITLE
All live radio channels play BBC World Service

### ISSUE THREAD

**Issue by dimkroon** [vonH/plugin.video.iplayerwww#364]
Justed tried v4.0.16, but every channel I tried to play ended up playing BBC World Service, often first displaying an error message.
I haven't got the time to look into it right now. This is just to let you know, I will add more details soon.

**Comment by CaptainTK** [comment]
I cannot duplicate that. Neither the error message nor all channels playing World Service.

However, live radio stopped working completely for me overnight. It now seems to need to allow for redirects. See e030ab3239a449410cd2676c7946e2ad53829218. Let me know if this fixes things for you as well. If so, I will release 4.0.17 asap.

**Comment by dimkroon** [comment]
I must say, it baffled me bit. It happened that I had Kodi setting player -> music -> play nest song automatically enabled. 
When Radio One failed it automatically tried the next, and so on until it reached BBC World, which plays without JWT. 
When I tested Radio One on OSMC, it did also try to play the next, but gave up before it got to BBC World. 

Yes, redirects fixes it. The access token must have expired overnight and are refreshed by redirects, just like OpenUrl() does.  In that respect, I opened #365  to suggest a little simplification.

While testing I noticed an unhandled exception when accessing geo-blocked content from outside of the UK. A fix is proposed is #366.

And I've finally brushed up some work I had lying around for some time, resulting in PR #367.
Apologies to spam you with all these PR's.
